src/sql/azdata.proposed.d.ts

```diff
@@ -432,6 +432,53 @@ declare module 'azdata' {
 		azurePortalEndpoint?: string;
 	}
 
+
+	export namespace diagnostics {
+		/**
+		 * Represents a diagnostics provider of accounts.
+		 */
+		export interface ErrorDiagnosticsProviderMetadata {
+			/**
+			 * The name of the provider (ex. a connection provider) that the diagnostics provider will handle errors for.
+			 */
+			id: string;
+		}
+
+		export interface ConnectionDiagnosticsResult {
+			/**
+			 * Status indicating if the error was handled or not.
+			 */
+			handled: boolean,
+			/**
+			 * The altered connection options after error handling.
+			 */
+			options: { [name: string]: any };
```

Comment:
This should be optional - for example if the error can be handled without changing the connection profile.